kube-monitoring/charts/values.yaml

```
@@ -197,6 +197,71 @@ kubeMonitoring:
       - services=[*]
       - statefulsets=[*]
 
+  ## Service discovery configuration
+  ##
+  serviceDiscovery:
+
+    ## Configuration for the Prometheus Operator to discover pods
+    ##
+    pods:
+      enabled: true
+
+      ## Endpoints of the selected pods to be monitored
+      ## https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#podmetricsendpoint
+      ##
+      podMetricsEndpoints:
+          ## Monitor Pods with the following port name
+        - port: metrics
+
+          ## RelabelConfigs to apply to samples before scraping
+          ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#relabelconfig
+          ##
+          relabelings: []
+          # - sourceLabels: [__meta_kubernetes_pod_node_name]
+          #   separator: ;
+          #   regex: ^(.*)$
+          #   targetLabel: nodename
+          #   replacement: $1
+          #   action: replace
+
+          ## MetricRelabelConfigs to apply to samples after scraping, but before ingestion.
+          ## ref: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#relabelconfig
+          ##
+          metricRelabelings: []
+          # - action: keep
+          #   regex: 'kube_(daemonset|deployment|pod|namespace|node|statefulset).+'
+          #   sourceLabels: [__name__]
+
+      ## Pod label for use in assembling a job name of the form <label value>-<port>
+      ## If no label is specified, the pod endpoint name is used.
+      ##
+      jobLabel: ""
+
+      ## Namespaces from which pods are selected
+      ##
+      namespaceSelector:
+        ## Match any namespace
+        ##
+        any: true
+
+        ## Explicit list of namespace names to select
+        ##
+        # matchNames: []
+
+      ## Example which selects all Pods to be monitored
+      ## with label "monitoredby" with values any of "example-pod-1" or "example-pod-2"
+      selector:
+        matchExpressions: []
```

The description is a copy paste error I assume.

Either we copy the [original](https://github.com/prometheus-community/helm-charts/blob/75bdfa0ed933adeacc452dabf3d40fbb01ba7cf1/charts/kube-prometheus-stack/values.yaml#L4426-L4434) or I would comment it out.

```suggestion
      ## Custom definition to pick up specific pods with an expression.
      # selector:
      ##  matchExpressions: []
```

---

This leads to a failed linting.  What is the error here?

---

Haven't run it, but then let's leave it in with a changed description.

```suggestion
      ## Custom definition to pick up specific pods with an expression.
      selector:
        matchExpressions: []
```